lessons/git2/slides.tex

```diff
@@ -0,0 +1,293 @@
+                    inner sep=1pt, minimum height=0.75em, minimum width=1em,
+                    font={\tiny\sffamily}},
+     connections/.style={draw}
+    ]
+    \node[anchor=west, font={\scriptsize}, align=center, xshift=1em] at (index.east)
+     (history) {History};
+    \matrix at (history.south)
+      [column sep={1em,between origins}, row sep={1em,between origins},
+       ampersand replacement=\&, yshift=-2.5em]%
+    {
+     \node[commit,visible on=<4->] (10) {\color{solarizedAccent} 10}; \\
+     \node[commit] (9) {9}; \\
+     \node[commit] (8) {8}; \\
+     \ghost{7}; \\
+    };
+    \path[connections, visible on=<4->] (9) -- (10);
+    \connect{8}{9};
+    \path[connections, densely dotted] (7) -- (8);
+   \end{scope}
+   \begin{scope}[
+     path/.style={->},
+     label/.style={decorate, decoration={text along path, text align=center,
+                                         text color=solarizedRebase0,
+                                         raise=1pt}}
+    ]
+    \draw[path] (git folder.east) to[out=0,in=180] (repository.west);
+    \draw (index.north west) rectangle (7.south -| history.east);
+    \draw<3->[decorate, decoration={brace, raise=0.75em, aspect=0.45}]
+      (first folder.east) |- (last folder.east) node[near start] (mid selection) {};
+    \draw<3->[path] (mid selection.east) + (0.6em,0) to[out=0,in=180] (snapshot.west);
+    \draw<4->[path] (snapshot.east) to[out=0,in=180] (10.west);
+   \end{scope}
+  \end{tikzpicture}
+  \pause
+  \item Start tracking a file with git (\texttt{git add})
+  \pause
+  \item Commit changes to the history (\texttt{git commit})
+  \pause
+  \item Check what's going on (\texttt{git status})
+  \item Compare a file with the one in the history (\texttt{git diff})
+  \item Look into your history (\texttt{git log})
+ \end{itemize}
+\end{frame}
+
+\section{Viewing History}
+
+\begin{frame}
+ \frametitle{Viewing History}
+
+ Viewing the log allows you to ``see'' history:
+ \begin{itemize}
+  \item \texttt{git log}
+  \pause
+  \item \texttt{git log \alert{<start>..<end>}}
+  \item \texttt{git log \alert{-{}- <file>}}
+  \item \texttt{git log \alert{-{}-oneline}}
+  \item \texttt{git log \alert{-{}-graph}}
+  \item \texttt{git log -{}-graph \alert{-{}-decorate}} \\
+ \end{itemize}
+ \pause
+ \begin{itemize}
+  \item \texttt{git blame <file>}
+ \end{itemize}
+ \pause
+ \begin{itemize}
+  \item \texttt{gitk} + \texttt{gitg} + other viewers
+ \end{itemize}
+\end{frame}
+
+\section{Branching}
+
+\begin{frame}
+ \frametitle{Branches}
+
+ What are branches?
+ \begin{itemize}[<+->]
+  \item Divergent commits (two commits with the same parent) could be
+        considered ``virtual'' branches
+  \item Branches are simply a \alert{named pointer} to a commit
+  \item Branches automatically \emph{move forward} as commits are made
+ \end{itemize}
+ \pause[\thebeamerpauses]
+ Why use them?
+ \pause
+ \begin{itemize}[<+->]
+  \item They're cheap!
```

Comment:
I added a little bit, though I try to avoid explaining _everything_ in my slides, or else why am I there to present it?